url7606378bcc_ljy2010a_protobuf2_tgz-pJ8-com_game_gameObj_bufBuilder_ReflectBuilderJ8/src/com/game/protobuf/GameProto.java

```diff
@@ -0,0 +1,7112 @@
+              return this;
+            default: {
+              if (!parseUnknownField(input, unknownFields,
+                                     extensionRegistry, tag)) {
+                this.setUnknownFields(unknownFields.build());
+                onChanged();
+                return this;
+              }
+              break;
+            }
+            case 10: {
+              com.game.protobuf.GameProto.UserInfo.Builder subBuilder = com.game.protobuf.GameProto.UserInfo.newBuilder();
+              if (hasUserInfo()) {
+                subBuilder.mergeFrom(getUserInfo());
+              }
+              input.readMessage(subBuilder, extensionRegistry);
+              setUserInfo(subBuilder.buildPartial());
+              break;
+            }
+          }
+        }
+      }
+      
+      private int bitField0_;
+      
+      // optional .protobuf.UserInfo userInfo = 1;
+      private com.game.protobuf.GameProto.UserInfo userInfo_ = com.game.protobuf.GameProto.UserInfo.getDefaultInstance();
+      private com.google.protobuf.SingleFieldBuilder<
+          com.game.protobuf.GameProto.UserInfo, com.game.protobuf.GameProto.UserInfo.Builder, com.game.protobuf.GameProto.UserInfoOrBuilder> userInfoBuilder_;
+      public boolean hasUserInfo() {
+        return ((bitField0_ & 0x00000001) == 0x00000001);
+      }
+      public com.game.protobuf.GameProto.UserInfo getUserInfo() {
+        if (userInfoBuilder_ == null) {
+          return userInfo_;
+        } else {
+          return userInfoBuilder_.getMessage();
+        }
+      }
+      public Builder setUserInfo(com.game.protobuf.GameProto.UserInfo value) {
+        if (userInfoBuilder_ == null) {
+          if (value == null) {
+            throw new NullPointerException();
+          }
+          userInfo_ = value;
+          onChanged();
+        } else {
+          userInfoBuilder_.setMessage(value);
+        }
+        bitField0_ |= 0x00000001;
+        return this;
+      }
+      public Builder setUserInfo(
+          com.game.protobuf.GameProto.UserInfo.Builder builderForValue) {
+        if (userInfoBuilder_ == null) {
+          userInfo_ = builderForValue.build();
+          onChanged();
+        } else {
+          userInfoBuilder_.setMessage(builderForValue.build());
+        }
+        bitField0_ |= 0x00000001;
+        return this;
+      }
+      public Builder mergeUserInfo(com.game.protobuf.GameProto.UserInfo value) {
+        if (userInfoBuilder_ == null) {
+          if (((bitField0_ & 0x00000001) == 0x00000001) &&
+              userInfo_ != com.game.protobuf.GameProto.UserInfo.getDefaultInstance()) {
+            userInfo_ =
+              com.game.protobuf.GameProto.UserInfo.newBuilder(userInfo_).mergeFrom(value).buildPartial();
+          } else {
+            userInfo_ = value;
+          }
+          onChanged();
+        } else {
+          userInfoBuilder_.mergeFrom(value);
+        }
+        bitField0_ |= 0x00000001;
+        return this;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 3246:3259, 3682:3695, 4118:4131.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.